### PR TITLE
bootstrap-tokens: clean up and separate signing and authentication

### PR DESCRIPTION
**Pull request by ericchiang**
NOTE: This is not a 1.9 specific PR. It can be merged whenever.

Was attempting to enable bootstrap token authentication on a cluster
and found this doc a little confusing for that sole purpose. This is a
small refactor that hopefully makes this document a little less `kubeadm`
specific.

This PR attempts to:
* Generally clean up the bootstrapping token doc.
* Separate the authentication and signing features. Help users that
  only want to enable the authenticator without the signer.
* Adds a warning about ConfigMap signing being susceptible to MITM
  if the same token is reused.

/assign @mattmoyer

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/website/6471)

### CODE REVIEW

File: docs/admin/bootstrap-tokens.md [hunk 1]
```diff
@@ -36,73 +36,84 @@ information.  It is used when referring to a token without leaking the secret
...
-group. `auth-extra-groups` indicates that it will also be included in the
-`system:bootstrappers:group1`, and `system:bootstrappers:group2` groups. The
-naming and groups are intentionally limited to discourage users from using these
-tokens past bootstrapping. Extra bootstrap token groups must start with
-`system:bootstrappers:`.
-
-`usage-bootstrap-signing` indicates that the token should be used to sign the
+* `usage-bootstrap-authentication` indicates that the token can be used to
+authenticate to the API server as a bearer token.
+* `usage-bootstrap-signing` indicates that the token should be used to sign the
```
Comment (tengqm) [inline docs/admin/bootstrap-tokens.md]: this is a "should" or a "may" ?

**Comment by ericchiang** [comment]
cc @kubernetes/sig-cluster-lifecycle-pr-reviews

**Comment by jbeda** [comment]
LGTM but would @mattmoyer to weigh in.

File: docs/admin/bootstrap-tokens.md [hunk 2]
```diff
@@ -156,3 +175,8 @@ is then used to form a whole JWS by inserting it between the 2 dots.  You can
 verify the JWS using the `HS256` scheme (HMAC-SHA256) with the full token (e.g.
 `07401b.f395accd246ae52d`) as the shared secret.  Users _must_ verify that HS256
 is used.
+
+WARNING: Any party with a bootstrapping token can create a valid signature for that
```
Comment (mattmoyer) [inline docs/admin/bootstrap-tokens.md]: There's a little section in the `kubeadm` docs that talks about this, not sure if it's worth linking or not: https://kubernetes.io/docs/reference/generated/kubeadm/#security-model

+1 on calling out this risk either way.

Comment (ericchiang) [reply docs/admin/bootstrap-tokens.md]: Didn't know that doc existed, thanks!

File: docs/admin/bootstrap-tokens.md [hunk 3]
```diff
@@ -116,6 +128,13 @@ In addition to authentication, the tokens can be used to sign a ConfigMap.  This
 is used early in a cluster bootstrap process before the client trusts the API
 server.  The signed ConfigMap can be authenticated by the shared token.
 
+Enable ConfigMap signing by enabling the `bootstrapsigner` controller on the
+Controller Manager.
+
+```
+--controllers=*,bootstrapsigner
```
Comment (luxas) [inline docs/admin/bootstrap-tokens.md]: Point out that you probably want to enable the tokencleaner then as well?

**Pull request merged**